### ISSUE TITLE
HalfEdgeTriangleMesh errors

### ISSUE THREAD

**Issue by kongsgard** [intel-isl/Open3D#1179]
**Describe the bug**
1) The example in `examples/Python/Basic/half_edge_mesh.py` does not work.
2) Converting the Stanford bunny given in `examples/TestData/Bunny.ply` from a TriangleMesh to a HalfEdgeTriangleMesh gives a segmentation fault.

**To Reproduce**
1) `python half_edge_mesh.py`
2)
```
mesh = o3d.io.read_triangle_mesh("Bunny.ply")
half_edge_mesh = o3d.geometry.HalfEdgeTriangleMesh.create_from_mesh(mesh)
```

**Expected behavior**
1) Find the single boundary, and display it in the visualizer.
2) Convert the TriangleMesh to a HalfEdgeTriangleMesh, which could be used for further processing such as e.g. boundary detection.

**Screenshots**
N/A

**Environment (please complete the following information):**
 - OS: OSX Mojave
 - Python version: 3.6.9
 - Open3D version: 0.8.0
 - Is this remote workstation?: no
 - How did you install Open3D?: conda

**Additional context**
1) I get the expected behavior if I change the name of the HalfEdgeTriangleMesh variable:
```
if __name__ == "__main__":
    mesh = o3d.io.read_triangle_mesh("../../TestData/sphere.ply")
    mesh = mesh.crop([-1, -1, -1], [1, 0.6, 1])
    half_edge_mesh = o3d.geometry.HalfEdgeTriangleMesh.create_from_mesh(mesh)
    mesh.compute_vertex_normals()
    num_vertices = len(mesh.vertices)
    draw_geometries_with_back_face([mesh])

    # Find a boundary vertex
    boundaries = half_edge_mesh.get_boundaries()
    assert len(boundaries) == 1
    boundary_vertices = boundaries[0]

    # Colorize boundary vertices
    vertex_colors = 0.75 * np.ones((num_vertices, 3))
    vertex_colors[boundary_vertices, :] = np.array([1, 0, 0])
    mesh.vertex_colors = o3d.utility.Vector3dVector(vertex_colors)
    draw_geometries_with_back_face([mesh])
```

2) If I add the same mesh cropping in the example above (although it doesn't seem to affect the mesh since the cropping is outside its boundaries), the conversion is successful. I can then also find the five holes in the mesh:
```
mesh = o3d.io.read_triangle_mesh('Bunny.ply')
mesh = mesh.crop([-1, -1, -1], [1, 0.6, 1])
half_edge_mesh = o3d.geometry.HalfEdgeTriangleMesh.create_from_mesh(mesh)
boundaries = half_edge_mesh.get_boundaries()
assert len(boundaries) == 5
```

**Comment by griegler** [comment]
The `HalfEdgeTriangleMesh` is unfortunately in a dis-functioning state at the moment. We will fix this asap.

**Comment by griegler** [comment]
Fixed in #1184.
